packages/devtools_app/lib/src/screens/memory/memory_filter.dart

```diff
@@ -186,7 +186,8 @@ class SnapshotFilterState extends State<SnapshotFilterDialog>
       // Don't include external and filtered these are a composite and can't be filtered out.
       if (library.name != externalLibraryName &&
           library.name != filteredLibrariesName) {
-        addLibrary(library.name);
+        assert(library.name != null);
```

Comment:
Is this assertion necessary?

Reply:
if name is null, next line will fail. We prefer assertion because we assume name is not null here.

Reply:
`library.name!` is effectively `assert(library.name != null)` though. It's not incorrect to assert as well, but it's a redundant check.